src/test/deploy/functions/params.spec.ts

```diff
@@ -0,0 +1,134 @@
+import * as chai from "chai";
+import * as sinon from "sinon";
+chai.use(require("chai-as-promised"));
+const expect = chai.expect;
+const prompt = require("../../../prompt");
+
+import * as params from "../../../deploy/functions/params";
+
+describe("resolveParams", () => {
+  let promptOnce: sinon.SinonStub;
+
+  beforeEach(() => {
+    promptOnce = sinon.stub(prompt, "promptOnce");
+  });
+
+  afterEach(() => {
+    promptOnce.restore();
+  });
+
+  it("can pull a literal value out of the dotenvs", async () => {
+    const paramsToResolve: params.Param[] = [
+      {
+        param: "foo",
+        type: "string",
+      },
+    ];
+    const userEnv: Record<string, string | number | boolean> = {
+      foo: "bar",
+    };
+    await expect(params.resolveParams(paramsToResolve, "", userEnv)).to.eventually.deep.equal({
+      foo: "bar",
+    });
+  });
+
+  it("errors when the dotenvs provide a value of the wrong type", async () => {
+    const paramsToResolve: params.Param[] = [
+      {
+        param: "foo",
+        type: "string",
+      },
+    ];
+    const userEnv: Record<string, string | number | boolean> = {
+      foo: 22,
+    };
+    await expect(params.resolveParams(paramsToResolve, "", userEnv)).to.eventually.be.rejected;
+  });
+
+  it("can use a provided literal", async () => {
+    const paramsToResolve: params.Param[] = [
+      {
+        param: "foo",
+        default: "bar",
+        type: "string",
+        input: { type: "text", text: {} },
+      },
+    ];
+    promptOnce.returns(Promise.resolve("bar"));
```

Comment:
I think you can say `promptOnce.resolves("bar")`

Reply:
You sure can.